### PR TITLE
Add support for user-supplied prefix/suffix in function declarations

### PR DESCRIPTION
**Pull request by kpet**
This change adds the ability for a user of the headers to pass
prefixes or suffixes to be used in the function prototypes via
(respectively) CL_API_PREFIX_USER and CL_API_SUFFIX_USER.

This can be used to pass additional attributes (e.g. to use
weak symbols).

Signed-off-by: Kévin Petit <kpet@free.fr>

### CODE REVIEW

**Comment by kpet** [comment]
Also see the (related) proposed tidy up on https://github.com/KhronosGroup/OpenCL-Headers/issues/153

File: CL/cl_platform.h [hunk 1]
```diff
@@ -41,22 +41,33 @@ extern "C" {
...
+#ifndef CL_API_SUFFIX_USER
+#define CL_API_SUFFIX_USER
+#endif
+
+#ifndef CL_API_PREFIX_USER
+#define CL_API_PREFIX_USER
+#endif
+
+#define CL_API_SUFFIX_COMMON CL_API_SUFFIX_USER
+#define CL_API_PREFIX_COMMON CL_API_PREFIX_USER
```
Comment (bashbaug) [inline CL/cl_platform.h:53]: I don't know how much we care about this, but I think we could avoid defining these symbols and just use the user-defined prefix and suffixes directly.  Alternatively, we could `#undef` these symbols after they're used.

As a separate issue (not for this PR), I think we should also formally reserve all of the `CL_API_XXX` names in the spec for usage in the headers.

**Review by bashbaug — APPROVED** [review]
LGTM, just one small comment / question that doesn't need to hold up merging.

File: CL/cl_platform.h (hunk 1 above)
Comment (kpet) [reply CL/cl_platform.h:53]: I added these to make it easier to manage common definitions in the future. I agree they're not strictly required for this change. We could indeed `#undef` them after but I'm not sure it matters too much, especially if we're reserving all the `CL_API*` names (I've added this to a checlist in #153).

**Comment by kpet** [comment]
I'll merge this one as is to get forward progress but I'm happy to discuss other changes as part of #153. I've broadened the scope of that issue to cover all related problems and added a checklist there so we don't lose track of what needs doing.

**Pull request merged**